pkg/blackduck/ctl_blackduck.go

```diff
@@ -265,161 +220,72 @@ func FlagWasSet(flagset *pflag.FlagSet, flagName string) bool {
-	baWasSet := flagset.Changed("enable-binary-analysis")
-	scAnalysisWasSet := flagset.Changed("enable-source-code-upload")
-	updatedBAEnviron := false
-	updatedSCAnalysisEnviron := false
-	for i, value := range valuesFromEnvironsFlag {
-		values := strings.SplitN(value, ":", 2)
-		mapKey := strings.TrimSpace(values[0])
-		if baWasSet && mapKey == "USE_BINARY_UPLOADS" {
-			if ctl.EnableBinaryAnalysis {
-				valuesFromEnvironsFlag[i] = "USE_BINARY_UPLOADS:1"
-			} else {
-				valuesFromEnvironsFlag[i] = "USE_BINARY_UPLOADS:0"
-			}
-			updatedBAEnviron = true
-		}
-		if scAnalysisWasSet && mapKey == "ENABLE_SOURCE_UPLOADS" {
-			if ctl.EnableSourceCodeUpload {
-				valuesFromEnvironsFlag[i] = "ENABLE_SOURCE_UPLOADS:true"
-			} else {
-				valuesFromEnvironsFlag[i] = "ENABLE_SOURCE_UPLOADS:false"
-			}
-			updatedSCAnalysisEnviron = true
-		}
-	}
-	if baWasSet && !updatedBAEnviron {
-		if ctl.EnableBinaryAnalysis {
-			valuesFromEnvironsFlag = append(valuesFromEnvironsFlag, "USE_BINARY_UPLOADS:1")
-		} else {
-			valuesFromEnvironsFlag = append(valuesFromEnvironsFlag, "USE_BINARY_UPLOADS:0")
-		}
-	}
-	if scAnalysisWasSet && !updatedSCAnalysisEnviron {
-		if ctl.EnableSourceCodeUpload {
-			valuesFromEnvironsFlag = append(valuesFromEnvironsFlag, "ENABLE_SOURCE_UPLOADS:true")
-		} else {
-			valuesFromEnvironsFlag = append(valuesFromEnvironsFlag, "ENABLE_SOURCE_UPLOADS:false")
-		}
-	}
-
-	// Overwrite any environs in the current spec with the new environ values from the flags (flag values take priority)
-	finalEnvironValues := util.MergeEnvSlices(valuesFromEnvironsFlag, ctl.blackDuckSpec.Environs)
-
-	// do not initialize on empty - unless it previously had values
-	if len(finalEnvironValues) > 0 || (len(finalEnvironValues) == 0 && len(ctl.blackDuckSpec.Environs) > 0) {
-		ctl.blackDuckSpec.Environs = finalEnvironValues
-	}
-	return nil
-}
-
 // SetCRSpecFieldByFlag updates a field in the blackDuckSpec if the flag was set by the user. It gets the
 // value from the corresponding struct field.
 // Note: It should only handle values with a 1 to 1 mapping - struct-field to spec
-func (ctl *CRSpecBuilderFromCobraFlags) SetCRSpecFieldByFlag(f *pflag.Flag) {
+func (ctl *HelmValuesFromCobraFlags) AddHelmValueByCobraFlag(f *pflag.Flag) {
 	if f.Changed {
 		log.Debugf("flag '%s': CHANGED", f.Name)
 		switch f.Name {
-		// case "enable-binary-analysis", "enable-source-code-upload", "environs": - these are handled in addEnvironsFlagValues()
 		case "size":
-			ctl.blackDuckSpec.Size = ctl.Size
-		case "version":
-			ctl.blackDuckSpec.Version = ctl.Version
-		case "expose-ui":
-			ctl.blackDuckSpec.ExposeService = ctl.ExposeService
-		case "db-prototype":
-			ctl.blackDuckSpec.DbPrototype = ctl.DbPrototype
-		case "external-postgres-host":
-			if ctl.blackDuckSpec.ExternalPostgres == nil {
-				ctl.blackDuckSpec.ExternalPostgres = &blackduckv1.PostgresExternalDBConfig{}
+			util.SetHelmValueInMap(ctl.args, []string{"size"}, ctl.Size)
```

Comment:
The BlackDuck Values.yaml doesn't have a `size` field

Reply:
Correct, since our Helm chart doesn't support sizes directly from the configuration file we need to have this hack in order to store the size somewhere. Synopsysctl is then responsible for reading the "size" value from the config and apply the patch.

Reply:
Oh gotcha! That's an interesting hack. Synopsysctl could use check the cmd.Flags("size").Value.ToString() and cmd.Flags("size").changed, but this works too :)

Reply:
The problem with only relying on the flag is that a user would have to specify the size every single time they update since the size wouldn't be stored anywhere

Reply:
Oh that's true! Okay I'm on board :)